lib/cocoapods-core/podfile/dsl.rb

```diff
@@ -273,32 +262,106 @@ def podspec(options = nil)
-      #             target :test, :exclusive => true do
+      #             target :ZipAppTests do
       #               pod 'JSONKit'
       #             end
       #           end
       #
       # @return   [void]
       #
-      def target(name, options = {})
-        if options && !options.keys.all? { |key| [:exclusive].include?(key) }
+      def target(name, options = nil)
+        if options
           raise Informative, "Unsupported options `#{options}` for " \
-            "target `#{name}`"
+            "target `#{name}`."
         end
 
         parent = current_target_definition
         definition = TargetDefinition.new(name, parent)
-        definition.exclusive = true if options[:exclusive]
         self.current_target_definition = definition
-        yield
+        yield if block_given?
       ensure
         self.current_target_definition = parent
       end
 
+      # Defines a new abstract target that can be used for convenient
+      # target inheritance.
+      #
+      # @param    [Symbol, String] name
+      #           the name of the target.
+      #
+      # @example  Defining an abstract target
+      #
+      #           abstract_target 'Networking' do
+      #             pod 'SSZipArchive'
+      #             target 'Networking App 1'
+      #             target 'Networking App 2'
+      #           end
+      #
+      # @return   [void]
+      #
+      def abstract_target(name)
+        target(name) do
+          abstract!
+          yield if block_given?
+        end
+      end
+
+      # Denotes that the current target is abstract, and thus will not directly
+      # link against an Xcode target.
+      #
+      # @return [void]
+      #
+      def abstract!(abstract = true)
+        current_target_definition.abstract = abstract
+      end
+
+      # Sets the inheritance mode for the current target.
+      #
+      # @param   [:complete, :none, :search_paths] the inheritance mode to set.
+      #
+      # @example Inheriting only search paths
+      #
+      #          target 'App' do
+      #            target 'AppTests' do
+      #              inherit! :search_paths
+      #            end
+      #          end
+      #
+      # @return  [void]
+      #
+      def inherit!(inheritance)
+        current_target_definition.inheritance = inheritance
+      end
+
+      # Specifies the installation method to be used when CocoaPods installs
+      # this Podfile.
+      #
+      # @param   [String] installation_method
+      #          the name of the installation strategy.
+      #
+      # @param   [Hash] options
+      #          the installation options.
+      #
+      # @example Specifying custom CocoaPods installation options
+      #
+      #          install! 'cocoapods',
+      #                   :deterministic_uuids => false,
+      #                   :integrate_targets => false
+      #
+      # @return  [void]
+      #
+      def install!(installation_method, options = {})
```

Comment:
Right now, it's only `cocoapods`. But in a future version, plugins will be able to register installation methods. And the options are validated inside the `InstallationOptions` class in `CocoaPods/CocoaPods`